manifest/sink.go

```diff
@@ -35,49 +52,168 @@ func (r *Reader) loadSinkConfig(pkg *pbsubstreams.Package, m *Manifest) error {
 		return fmt.Errorf("sink: config: error marshalling to json: %w", err)
 	}
 
-	r.sinkConfigJSON = string(jsonConfigBytes)
+	dynConf := dynamic.NewMessageFactoryWithDefaults().NewDynamicMessage(msgDesc)
+	if err := dynConf.UnmarshalJSON(jsonConfigBytes); err != nil {
+		return fmt.Errorf("sink: config: encoding json into protobuf message: %w", err)
+	}
+	r.sinkConfigDynamicMessage = dynConf
+	pbBytes, err := dynConf.Marshal()
+	if err != nil {
+		return fmt.Errorf("sink: config: encoding protobuf from dynamic message: %w", err)
+	}
+	pkg.SinkConfig = &anypb.Any{
+		TypeUrl: m.Sink.Type,
+		Value:   pbBytes,
+	}
+
+	return nil
+}
+
+func SinkConfigAsString(pkg *pbsubstreams.Package, dumpBytesToDisk bool) (string, error) {
```

Comment:
I dislike the fact that converting to string also dumps to file. In fact more reading lead me to think that this function is weirdly named.

It also deals with representational aspect, the output being a new-line separated string. The function should return a `[]string` and let the caller decide what to do.

I also think the path where to write things should be pass in by the caller. Inferring `tmp` seems unecessary, the caller should deal with this decision.